nav2-usability/scenario-code/dynamic-linking/dynamic_linking_vrouter.dart

```diff
@@ -0,0 +1,148 @@
+}
+
+class _WishlistAppState extends State<WishlistApp> {
+  final AppState _appState = AppState();
+  final vRouterKey;
+
+  _WishlistAppState() : vRouterKey = GlobalKey<VRouterState>() {
+    _appState.addListener(() {
+      setState(() {});
+    });
+  }
+
+  @override
+  void dispose() {
+    _appState.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return VRouter(
+      key: vRouterKey,
+      routes: [
+        VWidget(
+          path: '/',
+          widget: WishlistListScreen(wishlists: _appState.wishlists, onCreate: onCreate),
+          stackedRoutes: [
+            VWidget(
+              path: r'wishlist/:id(\d+)',
+              widget: Builder(
+                builder: (context) => WishlistScreen(
+                    wishlist: _appState.wishlists.firstWhere(
+                            (element) => element.id == context.vRouter.pathParameters['id']
+                    )),
+              ),
+            ),
+          ],
+        ),
+      ],
+    );
+  }
+
+  void onCreate(String value) {
+    final wishlist = Wishlist(value);
+    _appState.addWishlist(wishlist);
+    vRouterKey.currentState!.push('/wishlist/$value');
+  }
+}
+
+class WishlistListScreen extends StatelessWidget {
+  final List<Wishlist> wishlists;
+  final ValueChanged<String> onCreate;
+
+  WishlistListScreen({
+    required this.wishlists,
+    required this.onCreate,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: ListView(
+        children: [
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Text('Navigate to /wishlist/<ID> in the URL bar to dynamically '
+                'create a new wishlist.'),
+          ),
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: ElevatedButton(
+              onPressed: () {
+                var randomInt = Random().nextInt(10000);
+                onCreate('$randomInt');
+              },
+              child: Text('Create a new Wishlist'),
+            ),
+          ),
+          for (var i = 0; i < wishlists.length; i++)
+            ListTile(
+              title: Text('Wishlist ${i + 1}'),
+              subtitle: Text(wishlists[i].id),
+              onTap: () => context.vRouter.push('/wishlist/${wishlists[i].id}'),
+            )
+        ],
+      ),
+    );
+  }
+}
+
+class WishlistScreen extends StatelessWidget {
+  final Wishlist wishlist;
+
+  WishlistScreen({
+    required this.wishlist,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text('ID: ${wishlist.id}', style: Theme.of(context).textTheme.headline6),
+          ],
+        ),
+      ),
+    );
+  }
+}
```

Comment:
nit: new line at the end of file

Reply:
Done in https://github.com/flutter/uxr/pull/28/commits/c051ff928cbdad310b57c792bc63b720bef24e75.